proxy/src/main/java/net/md_5/bungee/BungeeCord.java

```diff
@@ -130,7 +129,7 @@
     /**
      * Fully qualified connections.
      */
-    private final Map<String, UserConnection> connections = new CaseInsensitiveMap<>();
+    private final Collection<UserConnection> connections = new HashSet<>();
```

Comment:
Please use a Multimap. We really do not want O(n) average performance on getting a player by name.

Reply:
I didn't take into account that bungee handles allot more players! I've re-written it to use a multimap now